File: backend/src/services/indexingservice.js
import { embeddings } from "../config/embeddings.js";
import { qdrant } from "../config/qdrant.js";
import {ensureCollection} from "./vectorservice.js"

export async function indexChunks({ chunks }) {
  await ensureCollection();  
  const vectors = await embeddings.embedDocuments(
    chunks.map(c => c.content)
  );
  console.log("Embedding length:", vectors[0]?.length);
  

  const points = chunks.map((chunk, i) => ({
    id: chunk.id,
    vector: vectors[i],
    payload: {
      documentId: chunk.documentId,
      userId: chunk.userId,
      order: chunk.order
    }
  }));

  await qdrant.upsert(process.env.QDRANT_COLLECTION, {
    points
  });

 
}

File: backend/src/services/indexing_service.py
from langchain_core.documents import Document

from src.db.models import Chunk
from src.integrations.vectorstore import get_vectorstore
from .vector_service import ensure_collection


def index_chunks(*, chunks: list[Chunk]) -> None:
    if not chunks:
        return

    ensure_collection()

    documents = [
        Document(
            page_content=chunk.content,
            metadata={
                "chunkId": chunk.id,
                "documentId": chunk.documentId,
                "userId": chunk.userId,
                "order": chunk.order,
            },
        )
        for chunk in chunks
    ]

    ids = [chunk.id for chunk in chunks]

    get_vectorstore().add_documents(
        documents=documents,
        ids=ids,
    )